app.rb

```diff
@@ -0,0 +1,119 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  def list_books
+    puts 'List of books:'
+    Book.all.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    puts 'List of people:'
+    Person.all.each do |person|
+      if person.respond_to?(:specialization)
+        print '[Teacher] '
+      else
+        print '[Student] '
+      end
+      puts "Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]:'
+    person_type = gets.chomp.to_i
+
+    case person_type
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Invalid option, please try again!'
+    end
+    puts 'Person created successfully!'
+  end
+
+  def create_student
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Has parent permission? [Y/N]:'
+    parent_permission = gets.chomp.downcase == 'y'
+    Student.new(age, name: name, parent_permission: parent_permission)
+  end
+
+  def create_teacher
+    puts 'Age:'
+    age = gets.chomp.to_i
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Specialization:'
+    specialization = gets.chomp
+    Teacher.new(age, name, specialization)
+  end
+
+  def create_book
+    puts 'Title:'
+    title = gets.chomp
+    puts 'Author:'
+    author = gets.chomp
+    Book.new(title, author)
+    puts 'Book created succesfully'
+  end
+
+  def create_rental
+    book = selected_book
+    person = selected_person
+    puts 'Date: '
+    date = gets.chomp
+    Rental.new(date, person, book)
+    puts 'Rental created succesfully'
+  end
+
+  def selected_book
+    puts 'Select a book from the following list by number'
+    Book.all.each_with_index do |book, index|
+      puts "(#{index}) Title: #{book.title}, Author: #{book.author}"
+    end
+    choice = gets.chomp.to_i
+    Book.all[choice]
+  end
+
+  def selected_person
+    puts 'Select a person from the following list by number (not id)'
+    Person.all.each_with_index do |person, index|
+      if person.respond_to?(:specialization)
+        puts "(#{index}) [Teacher] Name: #{person.name}, ID #{person.id}, Age: #{person.age}"
+      else
+        puts "(#{index}) [Student] Name: #{person.name}, ID #{person.id}, Age: #{person.age}"
+      end
+    end
+    choice = gets.chomp.to_i
+    Person.all[choice]
+  end
```

Comment:
- [ ] I was not able to create rentals at all because I couldn't access the ID of the people created. Perhaps because of the reason stated in the **previous review comment**. 

<img width="503" alt="Untitled" src="https://github.com/yodit93/school_library/assets/103386183/79c1fe67-e930-4a28-bd3d-ad314e91dc4d">

Reply:
Thanks, I will fix it.